managed-configuration-chrome.py

```diff
@@ -0,0 +1,115 @@
+configuration.api_key['Authorization'] = '<API-Key>'
+configuration.api_key_prefix['Authorization'] = 'Bearer'
+
+# create an instance of the API class
+api_instance = esperclient.CommandsV2Api(esperclient.ApiClient(configuration))
+enterprise_id = '<Enterprise ID>' # str | ID of the enterprise
+
+# Modify this Python Dict to set your own managed configurations
+command_args = CommandArgs(
+    custom_settings_config={
+        "managedAppConfigurations": {
+            "com.android.chrome": {
+                "URLBlacklist": ["https://www.omegle.com/","Facebook.com","instagram.com"], # not allow these websites to work
+                "URLWhitelist": ["*"], # rest all allowed to open
+                "IncognitoModeAvailability": "1",   # disable incognito mode
+                "ForceGoogleSafeSearch" : "true",   # enable safe search to work
+                "HomepageLocation": "https://esper.io"  # set chrome home page
+            }
+        }
+    }
+)
+
+# int | Number of results to return per page. (optional) (default to 20)
+configuration.per_page_limit = 5000
+# int | The initial index from which to return the results.(optional) default to 0)
+configuration.per_page_offset = 0
+
+allgroups = []
+
+# get all the groups id present in the enterprise and stores in allgroups[]
+def get_all_groups_in_enterprise():
+    # create an instance of the API class
+    api_instance = esperclient.DeviceGroupApi(esperclient.ApiClient(configuration))
+    try:
+        api_response = api_instance.get_all_groups(enterprise_id,
+                                                    limit=configuration.per_page_limit,
+                                                    offset=configuration.per_page_offset)
+        if len(api_response.results):
+            for group in api_response.results:
+                # add all the devices in this group to global list of devices
+                allgroups.append(group.id)
+        print(allgroups)
+
+    except ApiException as e:
+        print("Exception when calling DeviceGroupApi->get_all_groups: %s\n" % e)
+
+# V0CommandRequest | The request body to create a command for set of devices or groups
+## command_type ->
+##  * DEVICE: command request is meant for devices
+##  * GROUP: command request is meant for groups
+##  * DYNAMIC: command request is meant for dynamic set of devices i.e subset of devices from different groups or otherwise.
+## Type of devices to run commands on
+##  * active: Run commands on currently online devices
+##  * inactive: Run commands on currently offline devices
+##  * all: Run commands on all the devices. Commands will be queued for offline devices until they come back online.
+
+# contains the group id of all the groups in the enterprise
+def run_managed_configuration():
+
+    request = esperclient.V0CommandRequest(
+        enterprise=enterprise_id,
+        command_type="GROUP",
+        device_type="all",
+        groups = allgroups,
```

Comment:
all groups in a single command request will cause heavy load in the system. Pls try to create separate command requests for 1-3 groups.

Reply:
I have moved run_managed_configuration inside groups calculation and calling for each times